File: c++2025/Inheritance/Inheritance/Drink.cpp

#include "Drink.h"

//생성자 초기화 목록
Drink::Drink(string name, int price, int quantity) :
	name(name), price(price), quantity(quantity){ }

//금액 = 가격 x 수량
int Drink::calcPrice() {
	return price * quantity;
}

//주의: static을 붙이지 않음
void Drink::printTitle() {
	cout << "상품명\t가격\t수량\t금액\n";
}

void Drink::printData() {
	cout << name << "\t" << price << "\t"
		<< quantity << "\t" << calcPrice() << endl;
}

File: javaworks/Chapter7/src/salestatement/Drink.java
package salestatement;

public class Drink {
	//protected는 상속받는 클래스에서 접근 가능
	protected String name;  //상품명
	protected int price;    //가격
	protected int quantity;    //수량
	
	Drink(String name, int price, int quantity){
		this.name = name;
		this.price = price;
		this.quantity = quantity;
	}
	
	int calcPrice() {
		return price * quantity;  //금액 = 가격 * 수량
	}
	
	static void printTitle() { //제목 출력
		System.out.println("상품명\t가격\t수량\t금액");
	}
	
	void printData() { //데이터 출력
		System.out.println(name + "\t" + price + "\t" + 
							quantity + "\t" + calcPrice());
	}
}
